src/main/java/org/opensearch/security/http/OnBehalfOfAuthenticator.java

```diff
@@ -0,0 +1,249 @@
+        String rolesClaim = "";
+
+        if (er != null) {
+            rolesClaim = EncryptionDecryptionUtil.decrypt(encryptionKey, er.toString());
+        } else if (dr != null) {
+            rolesClaim = dr.toString();
+        } else {
+            log.warn("This is a malformed On-behalf-of Token");
+        }
+
+        List<String> roles = Arrays.stream(rolesClaim.split(","))
+            .map(String::trim)
+            .filter(s -> !s.isEmpty())
+            .collect(Collectors.toUnmodifiableList());
+
+        return roles;
+    }
+
+    private String[] extractBackendRolesFromClaims(Claims claims) {
+        Object backendRolesObject = claims.get("br");
+        String[] backendRoles;
+
+        if (backendRolesObject == null) {
+            log.warn("This is a malformed On-behalf-of Token");
+            backendRoles = new String[0];
+        } else {
+            // Extracting roles based on the compatibility mode
+            backendRoles = Arrays.stream(backendRolesObject.toString().split(",")).map(String::trim).toArray(String[]::new);
+        }
+
+        return backendRoles;
+    }
+
+    @Override
+    @SuppressWarnings("removal")
+    public AuthCredentials extractCredentials(RestRequest request, ThreadContext context) throws OpenSearchSecurityException {
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        AuthCredentials creds = AccessController.doPrivileged(new PrivilegedAction<AuthCredentials>() {
+            @Override
+            public AuthCredentials run() {
+                return extractCredentials0(request);
+            }
+        });
+
+        return creds;
+    }
+
+    private AuthCredentials extractCredentials0(final RestRequest request) {
+        if (!oboEnabled) {
+            log.error("On-behalf-of authentication is disabled");
+            return null;
+        }
+
+        if (jwtParser == null) {
+            log.error("Missing Signing Key. JWT authentication will not work");
+            return null;
+        }
+
+        String jwtToken = extractJwtFromHeader(request);
+        if (jwtToken == null) {
+            return null;
+        }
+
+        if (!isAllowedRequest(request)) {
+            return null;
+        }
+
+        try {
+            final Claims claims = jwtParser.parseClaimsJws(jwtToken).getBody();
+
+            final String subject = claims.getSubject();
+            if (subject == null) {
+                log.error("Valid jwt on behalf of token with no subject");
+                return null;
+            }
+
+            final String audience = claims.getAudience();
+            if (audience == null) {
+                log.error("Valid jwt on behalf of token with no audience");
+                return null;
+            }
+
+            final String issuer = claims.getIssuer();
+            if (!issuer.equals(clusterName)) {
```

Comment:
```suggestion
            if (!clusterName.equals(issuer)) {
```
clusterName is from a trusted source inside the security plugin whereas issuer is coming from the token, it could be null and you'd get a null reference, simpler to swap positions

Reply:
Fixed